classes/pdf/HTMLTemplateOrderSlip.php

```diff
@@ -67,11 +67,40 @@ public function __construct(OrderSlip $order_slip, Smarty $smarty)
         // header informations
         $this->date = Tools::displayDate($this->order_slip->date_add);
         $prefix = Configuration::get('PS_CREDIT_SLIP_PREFIX', Context::getContext()->language->id);
-        $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, (int) $this->order_slip->id);
+
+        $this->title = sprintf(
+            HTMLTemplateOrderSlip::l('%1$s%2$06d'),
+            $prefix,
+            Configuration::get('PS_CREDIT_SLIP_RESET') ? $this->getNumberOrderSlip((int) $order_slip->id) : (int) $order_slip->id
+        );
 
         $this->shop = new Shop((int) $this->order->id_shop);
     }
 
+    /**
+     * Returns the numbering of the subscription invoice with respect to the year.
+     *
+     * @param int $id_order_slip order_slip
+     *
+     * @return int number for year
+     */
+    public function getNumberOrderSlip(int $id_order_slip): int
+    {
+        $date = Db::getInstance()->getValue('SELECT date_add FROM `' . _DB_PREFIX_ . 'order_slip` WHERE id_order_slip=' . $id_order_slip);
+
+        if (isset($date)) {
+            $current_year = date('Y-01-01 00:00:00', strtotime($date));
+
+            $counts = Db::getInstance()->getValue(
+                'SELECT count(*) as "n" FROM `' . _DB_PREFIX_ . 'order_slip` WHERE date_add > "' . $current_year . '" AND date_add < "' . $date . '"'
+            );
+
+            return intval($counts) + 1;
+        }
+
+        return 1;
```

Comment:
```suggestion
        return 0;
```

Otherwise, it makes no sense to return 1 when the id_order_slip is not found :thinking: 
The number should be 0 has there is no data.

Reply:
It returns 1, because in the sql query it looks for a date less than the current invoice of the id_order_slip. So if there is only 1, the query will not return any items, therefore it is the first.

Reply:
So, the name of the method is not the good one :thinking:

Reply:
It would be more representative if it were named: _getNumberOfYearOrderSlip_?